### PR TITLE
feat: add alias option

### PR DESCRIPTION
**Pull request by sheremet-va**
Linked #1628

### CODE REVIEW

**Comment by antfu** [comment]
I am not so sure. I understand it could be easier for users to have env-specific alias, yet it might still be limited. I am concerned this would introduce some slippery slope effect that we might need to re-expose every vite option under `test`. Since we have `vitest.config` and `process.env.VITEST` on top of the fact that the config is js, I would prefer to reuse Vite's options as much as possible.

**Comment by sheremet-va** [comment]
> I am not so sure. I understand it could be easier for users to have env-specific alias, yet it might still be limited. I am concerned this would introduce some slippery slope effect that we might need to re-expose every vite option under `test`. Since we have `vitest.config` and `process.env.VITEST` on top of the fact that the config is js, I would prefer to reuse Vite's options as much as possible.

I think we should add `plugins` and `alias` support to `test` property, since they are highly requested, and it's easier to configure with a simple `test.alias/test.plugins` than go to the issues/discord and ask. Plugins are harder to implement right now, so I am waiting on multi repo support.

Anything else should be on Vite config. What do you think, @patak-dev?

**Comment by antfu** [comment]
OK, I am ok to have `alias` and `plugins` for the DX.

File: docs/config/index.md [hunk 1]
```diff
@@ -109,6 +109,12 @@ This might potentially cause some misalignment if a package has different logic
 
 Interpret CJS module's default as named exports.
 
+### alias
+
+- **Type:** `Record<string, string> | Array<{ find: string | RegExp, replacement: string, customResolver?: ResolverFunction | ResolverObject }>`
+
+Define custom aliases when running inside tests. They will be merged with aliases from `resolve.alias`.
```
Comment (belgattitude) [inline docs/config/index.md:116]: Could it be named something else than `alias` ? 

Genuinely asking cause it's not immediately clear to me how they compare/plug into the vite ones (`alias` in vite 2 and `resolve.alias` in vite 3). Easy to be confused when quickly checking config or looking for an existing issue.

Could it be renamed ? (ie: extraAlias, moduleMapper, testAlias...)

Comment (sheremet-va) [reply docs/config/index.md:116]: It’s on test.alias, what is confusing here?

Comment (belgattitude) [reply docs/config/index.md:116]: You're absolutely right, no confusion. My bad, I was confused :)

File: test/core/vitest.config.ts [hunk 2]
```diff
@@ -56,5 +56,13 @@ export default defineConfig({
     sequence: {
       seed: 101,
     },
+    alias: [
+      {
+        find: 'test-alias',
+        replacement: '',
+        // vitest doesn't crach because function is defined
```
Comment (sheremet-va) [inline test/core/vitest.config.ts:63]:
```suggestion
        // vitest doesn't crash because function is defined
```

**Pull request merged**